In [1]:
import pandas as pd
df = pd.read_csv("covid-19.csv")

### Review data frame

In [2]:
df.head()
df.head(5)

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
0    Afghanistan      36263    1269      25198    9796        106          10   
1        Albania       4880     144       2745    1991        117           6   
2        Algeria      27973    1163      18837    7973        616           8   
3        Andorra        907      52        803      52         10           0   
4         Angola        950      41        242     667         18           1   

   New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0             18                3.50                  69.49   
1             63                2.95                  56.25   
2            749                4.16                  67.34   
3              0                5.73                  88.53   
4              0                4.32                  25.47   

   Deaths / 100 Recovered  Confirmed last week  1 week change  \
0                    5.04                35526            737   
1                    5.25                 4171            709   
2                    6.17                23691           4282   
3                    6.48                  884             23   
4                   16.94                  749            201   

   1 week % increase             WHO Region  
0               2.07  Eastern Mediterranean  
1              17.00                 Europe  
2              18.07                 Africa  
3               2.60                 Europe  
4              26.84                 Africa

In [3]:
df.tail()
df.tail(5)

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  \
182  West Bank and Gaza      10621      78       3752    6791        152   
183      Western Sahara         10       1          8       1          0   
184               Yemen       1691     483        833     375         10   
185              Zambia       4552     140       2815    1597         71   
186            Zimbabwe       2704      36        542    2126        192   

     New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
182           2              0                0.73                  35.33   
183           0              0               10.00                  80.00   
184           4             36               28.56                  49.26   
185           1            465                3.08                  61.84   
186           2             24                1.33                  20.04   

     Deaths / 100 Recovered  Confirmed last week  1 week change  \
182                    2.08                 8916           1705   
183                   12.50                   10              0   
184                   57.98                 1619             72   
185                    4.97                 3326           1226   
186                    6.64                 1713            991   

     1 week % increase             WHO Region  
182              19.12  Eastern Mediterranean  
183               0.00                 Africa  
184               4.45  Eastern Mediterranean  
185              36.86                 Africa  
186              57.85                 Africa

In [4]:
#Changimg indexes to country/region
df = df.set_index("Country/Region")

In [5]:
df.columns
#View all columns

Index(['Confirmed', 'Deaths', 'Recovered', 'Active', 'New cases', 'New deaths',
       'New recovered', 'Deaths / 100 Cases', 'Recovered / 100 Cases',
       'Deaths / 100 Recovered', 'Confirmed last week', '1 week change',
       '1 week % increase', 'WHO Region'],
      dtype='object')

In [6]:
df.shape
#187 rows of 15 columns

(187, 14)

### Short statisctics

In [7]:
#get informations
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 187 entries, Afghanistan to Zimbabwe
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Confirmed               187 non-null    int64  
 1   Deaths                  187 non-null    int64  
 2   Recovered               187 non-null    int64  
 3   Active                  187 non-null    int64  
 4   New cases               187 non-null    int64  
 5   New deaths              187 non-null    int64  
 6   New recovered           187 non-null    int64  
 7   Deaths / 100 Cases      187 non-null    float64
 8   Recovered / 100 Cases   187 non-null    float64
 9   Deaths / 100 Recovered  187 non-null    float64
 10  Confirmed last week     187 non-null    int64  
 11  1 week change           187 non-null    int64  
 12  1 week % increase       187 non-null    float64
 13  WHO Region              187 non-null    object 
dtypes: float64(4), int64(9), object(

In [8]:
#short statistical
df.describe()

Confirmed         Deaths     Recovered        Active     New cases  \
count  1.870000e+02     187.000000  1.870000e+02  1.870000e+02    187.000000   
mean   8.813094e+04    3497.518717  5.063148e+04  3.400194e+04   1222.957219   
std    3.833187e+05   14100.002482  1.901882e+05  2.133262e+05   5710.374790   
min    1.000000e+01       0.000000  0.000000e+00  0.000000e+00      0.000000   
25%    1.114000e+03      18.500000  6.265000e+02  1.415000e+02      4.000000   
50%    5.059000e+03     108.000000  2.815000e+03  1.600000e+03     49.000000   
75%    4.046050e+04     734.000000  2.260600e+04  9.149000e+03    419.500000   
max    4.290259e+06  148011.000000  1.846641e+06  2.816444e+06  56336.000000   

        New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
count   187.000000     187.000000          187.000000             187.000000   
mean     28.957219     933.812834            3.019519              64.820535   
std     120.037173    4197.719635            3.454302              26.287694   
min       0.000000       0.000000            0.000000               0.000000   
25%       0.000000       0.000000            0.945000              48.770000   
50%       1.000000      22.000000            2.150000              71.320000   
75%       6.000000     221.000000            3.875000              86.885000   
max    1076.000000   33728.000000           28.560000             100.000000   

       Deaths / 100 Recovered  Confirmed last week  1 week change  \
count                  187.00         1.870000e+02     187.000000   
mean                      inf         7.868248e+04    9448.459893   
std                       NaN         3.382737e+05   47491.127684   
min                      0.00         1.000000e+01     -47.000000   
25%                      1.45         1.051500e+03      49.000000   
50%                      3.62         5.020000e+03     432.000000   
75%                      6.44         3.708050e+04    3172.000000   
max                       inf         3.834677e+06  455582.000000   

       1 week % increase  
count         187.000000  
mean           13.606203  
std            24.509838  
min            -3.840000  
25%             2.775000  
50%             6.890000  
75%            16.855000  
max           226.320000

### Look for those recovered from covid-19 (Patients)

In [9]:
#Total Recoverd from covid-19 in all countries
df["Recovered"].sum()

9468087

In [10]:
#highst recovered country is Brazil
df[df["Recovered"] == df["Recovered"].max()].iloc[:,2:3:]
# Brazill most recovered patients from covid-19

Recovered
Country/Region           
Brazil            1846641

In [11]:
#lowest recovered country is
df[df["Recovered"] == df["Recovered"].min()]["Recovered"].to_frame()

Recovered
Country/Region           
Canada                  0
Mozambique              0
Serbia                  0
Sweden                  0
Syria                   0
Timor-Leste             0

* Notice: There\`s no minimum recovery from covid-19 on those countries what does that mean? It means there\`s deaths in those countries

**Let\`s check**

In [12]:
#Check for deaths
df[df["Recovered"] <=0]["Deaths"].to_frame()

Deaths
Country/Region        
Canada            8944
Mozambique          11
Serbia             543
Sweden            5700
Syria               40
Timor-Leste          0

In [13]:
# #Get country`s Regions
df[df["Recovered"] <=0]["WHO Region"].to_frame()

WHO Region
Country/Region                       
Canada                       Americas
Mozambique                     Africa
Serbia                         Europe
Sweden                         Europe
Syria           Eastern Mediterranean
Timor-Leste           South-East Asia

In [14]:
#Comparison of Recoveries and Deaths
df[df["Recovered"] <=0 ].iloc[:,1:3:1]

Deaths  Recovered
Country/Region                   
Canada            8944          0
Mozambique          11          0
Serbia             543          0
Sweden            5700          0
Syria               40          0
Timor-Leste          0          0

In [15]:
#How many contries that has no recovery just deaths
df[df["Recovered"] <=0 ]["WHO Region"].count()
#About 6 countries had no recovery

6

* Calculate Total deaths of 6 countries

In [16]:
#Total Death for those 6 countries
df[df["Recovered"]<=0]["Deaths"].sum()

15238

* Check the rest of the countries that has death although they may have recovered (patients)

In [17]:
df[df["Recovered"] >0 ]["WHO Region"].count()
#About 181 Countries
df[df["Recovered"] >=0]["Deaths"].sum()
#181 country has death and recovered patients from covid-19

654036

### Look for highest deaths

In [18]:
#Total deaths occured
df["Deaths"].sum()

654036

In [19]:
df[df["Deaths"] == df["Deaths"].max()]
#US has the most country that had deaths occured by covid-19

Confirmed  Deaths  Recovered   Active  New cases  New deaths  \
Country/Region                                                                 
US                4290259  148011    1325804  2816444      56336        1076   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
US                      27941                3.45                   30.9   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
US                               11.16              3834677         455582   

                1 week % increase WHO Region  
Country/Region                                
US                          11.88   Americas

In [20]:
df[df["Deaths"] >0 ]["WHO Region"].count()
# 170 counties has deaths occured

170

In [21]:
df[df["Deaths"] <=0 ]["WHO Region"].count()
#The rest 17 counties has no deaths occured

17

In [22]:
#Get those countries
df[df["Deaths"] <=0 ]["WHO Region"].to_frame()

WHO Region
Country/Region                                   
Bhutan                            South-East Asia
Cambodia                          Western Pacific
Dominica                                 Americas
Eritrea                                    Africa
Fiji                              Western Pacific
Greenland                                  Europe
Grenada                                  Americas
Holy See                                   Europe
Laos                              Western Pacific
Mongolia                          Western Pacific
Papua New Guinea                  Western Pacific
Saint Kitts and Nevis                    Americas
Saint Lucia                              Americas
Saint Vincent and the Grenadines         Americas
Seychelles                                 Africa
Timor-Leste                       South-East Asia
Vietnam                           Western Pacific

***Comparision_1***

In [23]:
#17 countries had no deaths occured only recovered
df[df["Deaths"] <=0 ].iloc[::,1:3]

Deaths  Recovered
Country/Region                                     
Bhutan                                 0         86
Cambodia                               0        147
Dominica                               0         18
Eritrea                                0        191
Fiji                                   0         18
Greenland                              0         13
Grenada                                0         23
Holy See                               0         12
Laos                                   0         19
Mongolia                               0        222
Papua New Guinea                       0         11
Saint Kitts and Nevis                  0         15
Saint Lucia                            0         22
Saint Vincent and the Grenadines       0         39
Seychelles                             0         39
Timor-Leste                            0          0
Vietnam                                0        365

In [24]:
#6 countries had only deaths occured
df[df["Recovered"] <=0].iloc[::,1:3]

Deaths  Recovered
Country/Region                   
Canada            8944          0
Mozambique          11          0
Serbia             543          0
Sweden            5700          0
Syria               40          0
Timor-Leste          0          0

* 17 countries had Recovered only and 6 countries had only deaths, 23 as part of a total of 187 countries this means there\`re countires that had both recovered and deaths occured by covid-19

### New deaths and New recovered

#### New deaths

In [25]:
#Total new deaths
df["New deaths"].sum()

5415

In [26]:
df["New deaths"].min()
df[df["New deaths"] >0 ]["WHO Region"].count()
#96 countries has new deaths occured of the rest 91 countries

96

* The rest 91 countries had no new deaths

In [27]:
#Highest country that had new deaths
df[df["New deaths"] == df["New deaths"].max()]["WHO Region"].to_frame()

WHO Region
Country/Region           
US               Americas

***Insert table to get total deaths***

In [28]:
#Add new columns to calculate total deaths
df["Total deaths"] = df["Deaths"] + df["New deaths"]

In [29]:
#all deaths occured
df["Total deaths"].sum()

659451

***After calculate total death let\`s check the most death has occured by country***

In [30]:
df[df["Total deaths"] == df["Total deaths"].max()]#["WHO Region"].to_frame()
#US still the most deaths has occured by covid-19

Confirmed  Deaths  Recovered   Active  New cases  New deaths  \
Country/Region                                                                 
US                4290259  148011    1325804  2816444      56336        1076   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
US                      27941                3.45                   30.9   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
US                               11.16              3834677         455582   

                1 week % increase WHO Region  Total deaths  
Country/Region                                              
US                          11.88   Americas        149087

* Americas or US still had highest new deaths occured by covid-19

#### New recovered

In [31]:
#Total new recovery
df["New recovered"].sum()

174623

In [32]:
#Get the highest recovered country
df[df["New recovered"] == df["New recovered"].max()]
#Brazill has new recovered by 33,728

Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
Country/Region                                                                
Brazil            2442375   87618    1846641  508116      23284         614   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
Brazil                  33728                3.59                  75.61   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
Brazil                            4.74              2118646         323729   

                1 week % increase WHO Region  Total deaths  
Country/Region                                              
Brazil                      15.28   Americas         88232

In [33]:
df[df["New recovered"] == df["New recovered"].min()]["WHO Region"].count()
# 61 countries has no new recovered

61

***Insert table to calculate "Total recovery"***

In [34]:
df["Total recovered"] = df["Recovered"] + df["New recovered"]

In [35]:
#Total Recovered after adding new and old recovery
df["Total recovered"].sum()

9642710

***Look for the most total recovery by it\`s country***

In [36]:
df[df["Total recovered"] == df["Total recovered"].max()]
#Without selecting the exact country

Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
Country/Region                                                                
Brazil            2442375   87618    1846641  508116      23284         614   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
Brazil                  33728                3.59                  75.61   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
Brazil                            4.74              2118646         323729   

                1 week % increase WHO Region  Total deaths  Total recovered  
Country/Region                                                               
Brazil                      15.28   Americas         88232          1880369

* we see that Brazill has the most recovered from covid-19

In [37]:
df.loc["Brazil"]["Total recovered"]

1880369

In [38]:
#Check the minimun country recovery
df[df["Total recovered"] == df["Total recovered"].min()]["Total recovered"].to_frame()

Total recovered
Country/Region                 
Canada                        0
Mozambique                    0
Serbia                        0
Sweden                        0
Syria                         0
Timor-Leste                   0

In [39]:
df[df["Total recovered"] == df["Total recovered"].min()]["WHO Region"].count()
#At least 6 countries has no recovered yet

6

***Comparison_2*** 

In [40]:
df[df["Total recovered"] >0 ].iloc[:,14:16]

Total deaths  Total recovered
Country/Region                                   
Afghanistan                 1279            25216
Albania                      150             2808
Algeria                     1171            19586
Andorra                       52              803
Angola                        42              242
...                          ...              ...
West Bank and Gaza            80             3752
Western Sahara                 1                8
Yemen                        487              869
Zambia                       141             3280
Zimbabwe                      38              566

[181 rows x 2 columns]

In [41]:
#Those countries had no recovered just deaths
df[df["Total recovered"] <=0].iloc[::,14:16]

Total deaths  Total recovered
Country/Region                               
Canada                  8955                0
Mozambique                11                0
Serbia                   552                0
Sweden                  5703                0
Syria                     42                0
Timor-Leste                0                0

### Look for Activation to heal

In [42]:
#axis=0 == columns
df["Active"].sum(axis=0)
#Total activation

6358362

In [43]:
df[df["Active"] == df["Active"].max()]

Confirmed  Deaths  Recovered   Active  New cases  New deaths  \
Country/Region                                                                 
US                4290259  148011    1325804  2816444      56336        1076   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
US                      27941                3.45                   30.9   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
US                               11.16              3834677         455582   

                1 week % increase WHO Region  Total deaths  Total recovered  
Country/Region                                                               
US                          11.88   Americas        149087          1353745

* US had most active healings

In [44]:
df[df["Active"] > 0 ]["WHO Region"].count()
# 182 countries has Active process

182

***Calculate new cases***

In [45]:
df["New cases"].sum()

228693

In [46]:
df[df["New cases"] == df["New cases"].max()]
#US has maximum new cases

Confirmed  Deaths  Recovered   Active  New cases  New deaths  \
Country/Region                                                                 
US                4290259  148011    1325804  2816444      56336        1076   

                New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
Country/Region                                                             
US                      27941                3.45                   30.9   

                Deaths / 100 Recovered  Confirmed last week  1 week change  \
Country/Region                                                               
US                               11.16              3834677         455582   

                1 week % increase WHO Region  Total deaths  Total recovered  
Country/Region                                                               
US                          11.88   Americas        149087          1353745

* US still the highest country had cases to heal

***Insert "Total Cases activate"***

In [47]:
df["Total active cases"] = df["New cases"] + df["Active"]

In [51]:
#Total cases activations
df["Total active cases"].sum()

6587055

In [52]:
df[df["Total active cases"] == df["Total active cases"].max()]["WHO Region"].to_frame()
#Us still the highest

WHO Region
Country/Region           
US               Americas

In [53]:
df.groupby("Total active cases")["WHO Region"].max().to_frame()

WHO Region
Total active cases                 
0                   Western Pacific
1                   Western Pacific
2                            Europe
4                            Europe
8                            Europe
...                             ...
206704                       Europe
255115                       Europe
531400                     Americas
539956              South-East Asia
2872780                    Americas

[172 rows x 1 columns]

***Total confirmations***

In [54]:
df["Total confirmations"] = df["Confirmed"] + df["Confirmed last week"]

In [55]:
df["Total confirmations"].sum()

31194108

In [56]:
df[df["Total confirmations"] == df["Total confirmations"].max()]["WHO Region"].to_frame()

WHO Region
Country/Region           
US               Americas

In [60]:
#Save files
df.to_csv("modified_covid-19.csv")

### Summary Statistics
***comparison for the highest tow countries***

* **Brazil**
- total `deaths` 88,232
- total `recovered` 1,880,369
- total `cases active` 5,31,400
- total `confirmations` 4,561,021

* **US**
- total `deaths` 149,087
- total `recovered` 1,353,745
- total `active cases` 2,872,780
- total `confirmations` 8,124,936

***About "9,642,710" Total recovered from "187" countries and about "659,451" deaths occured by covid-19***

Currently cases healing about 6,587,055 patients as a total.
- The most country that has `deaths` occured is US="Anerican" which is about 149,087.
- The most country that has `active cases` to recover form covid-19 is US="American" which is about 2,872,780.
- The most country that has `recovered` occured is Brazil which is about 1,880,369.
- American has total `confirmations` booking about 8,124,936

### Decision report
**The best Country coud heal from covid-19**

***We see that `US` is the most country that has booking confirmations and related to it\`s cases active applied is highest although it has the most deaths occured, we decide `Brazil` is the most country you can rely on, to heal from covid-19 than `US`***